dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1058,4 +1058,31 @@ private void CaptureUsageDetails(CompletionsUsage usage)
         s_completionTokensCounter.Add(usage.CompletionTokens);
         s_totalTokensCounter.Add(usage.TotalTokens);
     }
+
+    /// <summary>
+    /// Processes the function result.
+    /// </summary>
+    /// <param name="functionResult">The result of the function call.</param>
+    /// <param name="toolCallBehavior">The ToolCallBehavior object containing optional settings like JsonSerializerOptions.TypeInfoResolver.</param>
+    /// <returns>A string representation of the function result.</returns>
+    private static string? ProcessFunctionResult(object functionResult, ToolCallBehavior? toolCallBehavior)
+    {
+        if (functionResult is string stringResult)
+        {
+            return stringResult;
+        }
+
+        // This is an optimization to use ChatMessageContent content directly  
+        // without unnecessary serialization of the whole message content class.  
+        if (functionResult is ChatMessageContent chatMessageContent)
+        {
+            return chatMessageContent.Content;
```

Comment:
nit: I'm wondering if we should use `chatMessageContent.ToString()` instead of using `Content` property directly. `ToString` override already use `Content` property, and in case if we would like to change the string representation of this type in the future by adding some additional properties, this place will be covered automatically:
https://github.com/microsoft/semantic-kernel/blob/8deaaf4e8285695f78d34360e8c7433f7da1d862/dotnet/src/SemanticKernel.Abstractions/Contents/ChatMessageContent.cs#L156-L159

Reply:
fixed